elementary/files
# Fix clear large folder

For folders with a large number of files (currently set to 1000, although that is open to change) when the model needs clearing (e.g. when changing from "show hidden" to "do not show hidden" or when navigating within the same view), instead of using the model·clear () function, which gets very slow for large folders, the model is destroyed and rebuilt.

Steps are taken to avoid unexpected change in the sort order when this done (also fixes bug in trunk where sort order may change on reload folder).

TO TEST:
(A)
1) Create a folder containing a large number of files (e.g ~100,000), using "split -b 100 SOME-10MB-FILE" or otherwise.
2) Show this folder (in ListView preferably for speed).
3) Monitor process cpu usage using gnome-system-monitor, htop or otherwise.
4) When cpu usage drops back to zero, switch on "Show Hidden".
5) When cpu usage drops back to zero, switch off "Show Hidden".

IN TRUNK:  When switching "Show Hidden" off, Files 100% of a cpu core for a significant length of time and may hang (or appear to).
IN BRANCH: When switching "Show Hidden" off, Files uses 100% of cpu core for a much shorter period of time and does not hang.

A similar difference is seen when reloading the folder.

(B)
1) Show two tabs with the same location and sort order (use ListView for ease).
2) Change the sort order in one tab.
3) Switch to other tab and press reload.

IN TRUNK:  The reloaded tab changes sort order.
IN BRANCH: The reloaded tab does not change sort order.
